### PR TITLE
[489] Executor sets the number of files for event data

### PR DESCRIPTION
**Pull request by giventocode**
Fixes: #489 

In this PR:
 - The executor sets the number of files for the event data correctly.
 - Added factory abstraction for the creation of uploader and downloaders inside executor, to faciliate testing. 
 - The test validates the correction.

### CODE REVIEW

**Review by BMurri — APPROVED** [review]

File: src/Tes.Runner.Test/ExecutorTests.cs [hunk 1]
```diff
@@ -79,6 +92,33 @@ public async Task DownloadInputsAsync_InputProvided_StartSuccessEventsAreCreated
...
+            };
+
+            blobDownloaderMock.Setup(r => r.DownloadAsync(It.IsAny<List<DownloadInfo>>()))
+                .ReturnsAsync(0);
+
+            fileOperationResolverMock.Setup(r => r.ResolveInputsAsync()).ReturnsAsync(inputs);
+            var result = await executor.DownloadInputsAsync(blobPipelineOptions);
+            Assert.AreEqual(Executor.ZeroBytesTransferred, result);
+            eventsPublisherMock.Verify(p => p.PublishDownloadStartEventAsync(It.IsAny<NodeTask>()), Times.Once);
+            eventsPublisherMock.Verify(p => p.PublishDownloadEndEventAsync(It.IsAny<NodeTask>(), nodeTask.Inputs.Count, 0, EventsPublisher.SuccessStatus, string.Empty), Times.Once);
```
Comment (MattMcL4475) [inline src/Tes.Runner.Test/ExecutorTests.cs:119]: Maybe add another test or modify this one, to test a non-zero number of inputs and assert the download count, since that was the original bug?

**Review by MattMcL4475 — APPROVED** [review]

File: src/Tes.Runner.Test/ExecutorTests.cs (hunk 1 above)
Comment (giventocode) [reply src/Tes.Runner.Test/ExecutorTests.cs:119]: Hmm this is what this test is asserting. It verifies that the event is created with the correct number of files (nodeTask.Inputs.Count).
